src/index.ts

```diff
@@ -109,14 +111,80 @@ export const bootstrapEmpackPackedEnvironment = async (
         });
 
         sharedLibsMap[pkg.name] = getSharedLibs(extractedPackage, '');
-
         saveFilesIntoEmscriptenFS(Module.FS, extractedPackage, '');
       })
     );
     await waitRunDependencies(Module);
   }
 
-  return sharedLibsMap;
+  return { sharedLibs: sharedLibsMap, untarjs };
+};
+
+export interface IRemovePackagesFromEnvOptions {
+  /**
+   * The list of packages which should be removed
+   */
+  removeList: any;
+
+  /**
+   * The URL (CDN or similar) from which to download packages
+   */
+  pkgRootUrl: string;
+
+  /**
+   * The Emscripten Module
+   */
+  Module: any;
+
+  /**
+   * Whether to install conda-meta for packages, default to False
+   */
+  generateCondaMeta?: boolean;
+
+  /**
+   * The untarjs API
+   */
+  untarjs: IUnpackJSAPI;
+
+  /**
+   * The logger to use during the bootstrap.
+   */
+  logger?: ILogger;
+}
+
+/**
+ * Removing previously installed files
+ *
+ * @param options
+ * @returns void
+ */
+export const removingFiles = async (
+  options: IRemovePackagesFromEnvOptions
+): Promise<void> => {
+  const { removeList, pkgRootUrl, Module, generateCondaMeta, logger, untarjs } =
+    options;
+
+  const pythonVersion = getPythonVersion(removeList);
+
+  if (removeList.length) {
+    await Promise.all(
+      removeList.map(async pkg => {
+        const url = pkg?.url ?? `${pkgRootUrl}/${pkg.filename}`;
+        logger?.log(`Removing ${pkg.filename}`);
+
+        const extractedPackage = await untarCondaPackage({
+          url,
+          untarjs,
+          verbose: false,
+          generateCondaMeta,
+          pythonVersion
+        });
+
+        removeFilesFromEmscriptenFS(Module.FS, extractedPackage, '');
```

Comment:
It's a bit unfortunate that we'd need to download and untar the package again just to get filenames information for removing them from the FS.

Can we instead store in-memory a dictionary mapping of `packages name` -> `file paths` so we know what to remove?